lib/src/main/java/at/bitfire/cert4android/UserDecisionRegistry.kt

```diff
@@ -108,10 +108,17 @@ class UserDecisionRegistry private constructor(
             customCertStore.setUntrustedByUser(cert)
 
         // continue work that's waiting for decisions
-        pendingDecisions[cert]?.iterator()?.let { iter ->
-            while (iter.hasNext()) {
-                iter.next().resume(trusted)
-                iter.remove()
+        synchronized(pendingDecisions) {
+            pendingDecisions[cert]?.iterator()?.let { iter ->
+                while (iter.hasNext()) {
+                    try {
+                        iter.next().resume(trusted)
+                    } catch(_: IllegalStateException) {
+                        Cert4Android.log.warning("Tried to resume an already resumed continuation.")
```

Comment:
Hm I'm not that happy with a "blind" catch when we don't know what's it actually for. I think we should put more efforts in finding out the cause of the problem and make sure that the continuation can't be resumed twice. If we can't guarantee that, continuation may not be the right way to do it and we should use `CompleteableDeferred` or whatever else.